src/order.py

```diff
@@ -1,169 +1,44 @@
-        # self.__dict__ + extend by missing bits
-        json_str = '''{
-          "sellToken": "''' + str(self["sellToken"]) + '''",
-          "buyToken": "''' + str(self["buyToken"]) + '''",
-          "receiver": "''' + str(self["receiver"]) + '''",
-          "sellAmount": "''' + str(self["sellAmount"]) + '''",
-          "buyAmount": "''' + str(self["buyAmount"]) + '''",
-          "validTo": ''' + str(self["validTo"]) + ''',
-          "appData": "''' + str(self["appData"].hex()) + '''",
-          "feeAmount": "''' + str(self["feeAmount"]) + '''",
-          "kind": "''' + str(self["kind"]) + '''",
-          "partiallyFillable": ''' + str(self["partiallyFillable"]).lower() + ''',
-          "signature": "''' + str(signed_message.signature.hex()) + '''",
-          "signingScheme": "ethsign",
-          "sellTokenBalance": "''' + str(self["sellTokenBalance"]) + '''",
-          "buyTokenBalance": "''' + str(self["buyTokenBalance"]) + '''",
-          "from": "''' + public_key + '''"
-        }'''
-        return json.loads(json_str)
-
-    def place_order(self):
-        self.fee_amount = self.get_fee()
-        order_bytes = self.signable_bytes(domain)
-        order_hash = w3.keccak(order_bytes)
-        message = encode_defunct(primitive=order_hash)
-        signed_message = w3.eth.account.sign_message(message, private_key=private_key)
-        # Build Json out of Order + signed_message.signature.hex() + public_address
-        order_json = self.to_json(signed_message, public_address)
-        r = requests.post(base_url + "orders", json=order_json)
-        # TODO - log status code.
-
-    def counter_order(self):
-        # TODO - stop accessing fields by string keys.
-        res = Order(
-            sell_token=self.buy_token,
-            buy_token=self.sell_token,
-            receiver=public_address,
-            sell_amount=self.buy_amount,
-            buy_amount=self.sell_amount,
-            # 2 minutes in the future.
-            valid_to=int(time.time()) + 120,
-            kind=self.opposite_kind(),
-        )
-        res.fee_amount = res.get_fee()
-        # TODO - could make method set_fee which updates field directly.
-        return res
+import Domains
+import json
+import requests
+from JSONEncoder import BytesJSONEncoder
+
+class Order(EIP712Message):
+    _name_: "string" = Domains.name
+    _version_: "string" = Domains.version
+    _chainId_: "uint256" = Domains.chainId
+    _verifyingContract_: "address" = Domains.verifyingContract
+
+    sellToken: "address"
+    buyToken: "address"
+    receiver: "address" = Domains.public_address1
+    sellAmount: "uint256"
+    buyAmount: "uint256"
+    validTo: "uint32" = int(int(time.time()) + 240)
+    appData: "bytes32" = HexBytes("0x0000000000000000000000000000000000000000000000000000000000000ccc")
+    feeAmount: "uint256" 
+    kind: "string"
+    partiallyFillable: "bool" = False
+    sellTokenBalance: "string" = "erc20"
+    buyTokenBalance: "string" = "erc20"
```

Comment:
In the previous version of the Order class we had set the field types imported from the eip712_structs library. Namely `Address, Boolean, Bytes, String` and `Uint`. Is there some reason these types are now replaced with strings?

Reply:
Yeah I used another python eip712 wrapper because I found it a bit more Pythonic